extras/data/machinery/generate/utils.py

```diff
@@ -81,14 +81,39 @@ def dump_polarized(src_path, target):
         observables dictionary, corresponding to the runcard
 
     """
+
+    # We make a case distinction for several experiments:
     obs = obs_template.copy()
-    data = load(src_path, 0, ["x", "Q2"])
-    dict_kins = [
-        dict(x=d["x"]["mid"], y=d["y"]["mid"], Q2=d["Q2"]["mid"]) for d in data
-    ]
+    if "ATHENA" in target.parent.name:
+        data = np.genfromtxt(src_path)
+        dict_kins = [dict(x=float(d[0]), y=float(d[2]), Q2=float(d[1])) for d in data]
+
+        obs["PolarizationDIS"] = 0.0 if "_F1" in target.parent.name else 1.0
+        observable_name = "F1_total" if "_F1" in target.parent.name else "g1_total"
+
+    elif "EIC" in target.parent.name:
+        data = np.genfromtxt(src_path)
+        dict_kins = [dict(x=float(d[0]), y=0.0, Q2=float(d[1])) for d in data]
+
+        obs["PolarizationDIS"] = 0.0 if "_F1" in target.parent.name else 1.0
+        observable_name = "F1_charm" if "_F1" in target.parent.name else "g1_charm"
+
+    elif "EIcC" in target.parent.name:
+        data = np.genfromtxt(src_path)
+        dict_kins = [dict(x=float(d[0]), y=0.0, Q2=float(d[1])) for d in data]
+
+        obs["PolarizationDIS"] = 0.0 if "_F1" in target.parent.name else 1.0
+        observable_name = "F1_charm" if "_F1" in target.parent.name else "g1_charm"
+
+    else:
+        data = load(src_path, 0, ["x", "Q2"])
+        dict_kins = [
+            dict(x=d["x"]["mid"], y=d["y"]["mid"], Q2=d["Q2"]["mid"]) for d in data
+        ]
+
+        obs["PolarizationDIS"] = 0.0 if "_F1" in target.parent.name else 1.0
+        observable_name = "F1_total" if "_F1" in target.parent.name else "g1_total"
```

Comment:
nice try, but no! okok, I admit it - this is already better than the last time, but not good enough yet :upside_down_face: there is still a lot of repetition and together we can get rid of it! Let's start from the end:
- 4th line: how about taking `heavyness` as argument to the function? default it to `"total"` and you save 75-5 character of repetition
- 3rd line: 69 character of repetition
- 1st and 2nd line: fine we need to think a tiny bit more - but we are developers, we can do it! :upside_down_face: e.g. we can simply harmonize the file format: i.e. parse the EIC kinematics file and make them compatible with the others. Another possibility would be to split the function and eventually take `dict_kins` as input, or many possibilities more - the Italians say at this point: "Tutte le strade portano a Roma" (which you may recognize from Asterix)

Don't repeat yourself.